src/sbvr-api/sbvr-utils.ts

```diff
@@ -486,30 +529,42 @@ export const executeModels = async (
 				await syncMigrator.run(tx, model);
 				const compiledModel = generateModels(model, db.engine);
 
-				// Create tables related to terms and fact types
-				// Run statements sequentially, as the order of the CREATE TABLE statements matters (eg. for foreign keys).
-				for (const createStatement of compiledModel.sql.createSchema) {
-					const promise = tx.executeSql(createStatement);
-					if (db.engine === 'websql') {
-						promise.catch((err) => {
-							console.warn(
-								"Ignoring errors in the create table statements for websql as it doesn't support CREATE IF NOT EXISTS",
-								err,
-							);
-						});
+				if (compiledModel.sql) {
+					// Create tables related to terms and fact types
+					// Run statements sequentially, as the order of the CREATE TABLE statements matters (eg. for foreign keys).
+					for (const createStatement of compiledModel.sql.createSchema) {
+						const promise = tx.executeSql(createStatement);
+						if (db.engine === 'websql') {
+							promise.catch((err) => {
+								console.warn(
+									"Ignoring errors in the create table statements for websql as it doesn't support CREATE IF NOT EXISTS",
+									err,
+								);
+							});
+						}
+						await promise;
 					}
```

Comment:
This loop will spawn promises after each other but they must not executed after each other. If we surely want to have them executed after each other the next iteration should wait until the promise resolved: 
```suggestion
					for await (const createStatement of compiledModel.sql.createSchema) {
						const promise = tx.executeSql(createStatement);
						if (db.engine === 'websql') {
							promise.catch((err) => {
								console.warn(
									"Ignoring errors in the create table statements for websql as it doesn't support CREATE IF NOT EXISTS",
									err,
								);
							});
						}
						await promise;
					}
```

Reply:
The current behavior is correct where they are executed in order, `for await...of` is for the case where it's an async iterator, ie if we did
```
for await (const x of asyncIterable) {
```
where `asyncIterable` has a `Symbol.asyncIterator` property that returns an iterator that returns promises, for example a paginated api where you might do
```
for await (const page of paginatedApiResponse) {
```
where each loop fetches the next page asynchronously.

Also check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of for more details/examples

Reply:
Oh I see, thanks and sorry for the confusion.